internals/overlord/stateengine.go

```diff
@@ -72,12 +80,38 @@ func (se *StateEngine) State() *state.State {
 	return se.state
 }
 
-type ensureError struct {
+type multiError struct {
 	errs []error
 }
 
-func (e *ensureError) Error() string {
-	return fmt.Sprintf("state ensure errors: %v", e.errs)
+func (e *multiError) Error() string {
+	errStrings := make([]string, len(e.errs))
+	for i, err := range e.errs {
+		errStrings[i] = err.Error()
+	}
+	return "multiple errors: " + strings.Join(errStrings, "; ")
+}
+
+// DryStart ensures all managers are ready to run their activities
+// before the first call to Ensure() is made.
+func (se *StateEngine) DryStart() error {
+	se.mgrLock.Lock()
+	defer se.mgrLock.Unlock()
+	if se.stopped {
+		return ErrStateEngineStopped
+	}
+	var errs []error
+	for _, m := range se.managers {
+		err := m.DryStart()
+		if err != nil {
+			logger.Noticef("dry start error: %v", err)
```

Comment:
Do we need to log this in addition to to returning the errors? It seems like whoever's handling the dry start should already check that (multi)error.